File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<int[]> minheap=new PriorityQueue<>((a,b)->a[0]-b[0]);
        for(int[] x:points){
            int is = x[0] * x[0] + x[1] * x[1];
            minheap.offer(new int[]{is,x[0],x[1]});
        }
        int[][] res=new int[k][2];
        for(int i=0;i<k;i++){
            int[] y=new int[3];
            y=minheap.poll();
            res[i][0]=y[1];
            res[i][1]=y[2];

        }
        
        return res;
    }
}

File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.py
class Solution(object):
    def kClosest(self, points, k):
       def helper(points,k):
        heap=[]
        for (x,y) in points:
            ios=x*x + y*y
            heapq.heappush(heap,(ios,[x,y]))
        res=[]
        for _ in range(k):
            res.append(heapq.heappop(heap)[1])
        return res
       return helper(points,k)

        